tests/foreman/api/test_computeresource_vmware.py

```diff
@@ -194,7 +194,7 @@ def test_positive_provision_vmware_pxe_discovery(
     vmware_host.start()
     wait_for(
         lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
-        timeout=1500,
```

Comment:
Increasing timeout doesn't work always. There was an issue with dhcp leases which causes this. I am trying to fix all discovery related issues in https://github.com/SatelliteQE/robottelo/pull/18297 and https://github.com/SatelliteQE/robottelo/pull/18265

Reply:
If mentioned issues fix the same problem, then we can close this PR?
If not, let's continue reviewing...
wait_for does not seem to support _retries_ parameter - see https://github.com/RedHatQE/wait_for. I would recommend removing it, because it looks confusing together with _timeout_ and _retries_.